docs/decisions/0025-dsp-pipeline-architecture.md

```diff
@@ -0,0 +1,524 @@
+| -------------------- | ---------------------------- | ------------------------- | ------------------------ | --------------------- |
+| **Throughput**       | 10+ GB/s (zero-copy)         | ~200 MB/s (copies)        | Same as Option 1         | 50+ GB/s              |
+| **Latency**          | <0.1ms (SAB) + ~1ms (worker) | ~3-5ms total              | ~1ms                     | ~2-5ms (GPU transfer) |
+| **Scalability**      | Excellent (N workers)        | Good (N workers)          | Poor (1 worker)          | Excellent             |
+| **Browser Support**  | Modern only (92%+)           | All (99%+)                | All                      | Limited (70%)         |
+| **Deployment**       | Requires COOP/COEP           | Simple                    | Simple                   | Simple                |
+| **Debugging**        | Moderate (multi-thread)      | Moderate                  | Easy                     | Hard                  |
+| **WASM Integration** | Excellent                    | Good                      | Good                     | N/A                   |
+| **Suitable for**     | High-perf real-time          | General purpose           | Prototyping              | Extreme perf          |
+
+## More Information
+
+### Related ADRs
+
+- [ADR-0002: Web Worker DSP Architecture](./0002-web-worker-dsp-architecture.md) - Initial worker architecture
+- [ADR-0012: Parallel FFT Worker Pool](./0012-parallel-fft-worker-pool.md) - Worker pool strategies
+- [ADR-0015: SIMD DSP Optimization](./0015-simd-dsp-optimization.md) - WASM SIMD acceleration
+
+### Reference Documentation
+
+- [DSP Processing Pipeline](../dsp-processing-pipeline.md) - API documentation
+- [SharedArrayBuffer Ring Buffer](../../src/utils/sharedRingBuffer.ts) - Implementation
+- [WASM DSP Functions](../../assembly/dsp.ts) - WASM implementations
+
+### Performance Benchmarks
+
+- Baseline: M1 MacBook Pro, Chrome 120
+- SharedRingBuffer: 10.2 GB/s write, 9.8 GB/s read (benchmarked)
+- WASM FFT (2048 points): ~0.3ms (7x faster than Web Audio API)
+- WASM Windowing (2048 points): ~0.05ms (3x faster than JavaScript)
+
+### Browser Compatibility
+
+- **Chrome/Edge**: 92+ (July 2021)
+- **Firefox**: 79+ (July 2020)
+- **Safari**: 15.2+ (December 2021)
+- **Mobile**: iOS 15.2+, Android Chrome 92+
+
+### Security Considerations
+
+SharedArrayBuffer was initially disabled due to Spectre/Meltdown vulnerabilities. The COOP/COEP headers requirement ensures:
+
+1. **Isolation**: Page cannot be embedded in other origins
+2. **No sharing**: Subresources must opt-in via CORP
+3. **Attack surface**: Reduced side-channel attack vectors
+
+### Future Enhancements
+
+1. **Adaptive quality**: Degrade FFT size when CPU saturated
+2. **WebGPU compute**: Upgrade FFT to GPU compute shader (Option 4)
+3. **Rust+WASM**: Alternative to AssemblyScript for more complex DSP
+4. **Audio Worklets**: Move demodulation to Audio Worklet for lower latency
```

Comment:
Should we be doing all demodulation in Audio Worklets, or is our WASM implementation better?